connectors/src/connectors/intercom/lib/conversation_permissions.ts

```diff
@@ -51,6 +54,30 @@ export async function allowSyncTeam({
         name: teamOnIntercom.name,
         permission: "read",
       });
+
+      // Update datasource folder node
+      const connector = await ConnectorResource.fetchById(connectorId);
+      if (connector !== null) {
+        // Create folder node for the team
+        const teamInternalId = getTeamInternalId(
+          connectorId,
+          teamOnIntercom.id
+        );
+        await upsertDataSourceFolder({
+          dataSourceConfig: dataSourceConfigFromConnector(connector),
```

Comment:
I think we want to do these in the activity and not at a place where the user would wait for it

Reply:
:+1: